lib/backends/vault-backend.js

```diff
@@ -52,8 +52,11 @@ class VaultBackend extends KVBackend {
 
     this._logger.debug(`reading secret key ${key} from vault`)
     const secretResponse = await this._client.read(key)
-
-    return JSON.stringify(secretResponse.data.data)
+    /**As the stringify adds addtional double quotes to the data that is retrieved 
+    from the Vault in JSON format we need to parse it to have actual data rather than
+    having a double quotes added data content.*/
+    secretValue = JSON.stringify(secretResponse.data.data)
+    return JSON.secretdataVaule(secretVaule)
```

Comment:
This line doesn't look right

Reply:
Have fixed it please review it. Thanks

Reply:
@Flydiverny : Can you please review I have fixed it.